### PR TITLE
vim-patch:8.1.1726,8.2.{296,860,1827,2388,2788,2790,2801}

### PR DESCRIPTION
**Pull request by janlazo**
**Please report any syntax or API issues with the runtime files to their respective maintainers.**

I'm going through all 70 runtime patches. Expect tons of breaking changes, at least for documentation. My aim is to port runtime patches before patch v8.2.0149 (vim9) by next week.

I'd appreciate assistance on the po files.
I haven't removed the duplicate translations yet so I expect all builds to fail.

Omit (or partially port) the following:
- translations: de.po, ru.po, any po file that require duplicates for full port
- features, syntax, and documentation for unmerged patches (ie. state(), vim9, modifyOtherKeys)
- Vim's textprop, terminal, popupwin
- GUI (ie. `scrollfocus`)
- tutor
- vi_diff.txt (should be N/A because Neovim has vim_diff.txt and removed most references to Vi)
  - didn't notice that it's actually in the repo. whoops :facepalm: 
- desktop files (ie. gvim.desktop)
- version8.txt
- old plugins and tools (ie. logipat, getscripts)
- `.github/` files (ie. CODEOWNERS)
  - handle in separate issue and PR, not this one

20 patches done after syncing most runtime files with Vim 8.2.0.
41 patches done after syncing up before Vim 8.2.0930.
~~50+ to go towards Vim master.~~
<10 left to go towards Vim master.

### CODE REVIEW

**Comment by janlazo** [comment]
:relieved: Finally done (today). Most of the runtime commits aren't complete ports but the same goes for the vim-patch PRs since vim9.

Going to merge tonight (if I don't lose internet or electricity) if no one volunteers to review. It's shouldn't difficult to review if one trusts me to not mess up copy/paste when vim-patch.sh doesn't auto-apply the patch on the file. I actually messed up on some of them but all should be fixed by the last commit. Majority of the review should be spent on the runtime files, maintained by Vim, because I modified those files to omit vim9 or port to pre-8.2 Vimscript syntax (ie. method).

**Review by jamessan — COMMENTED** [review]
Didn't do a thorough review, but tried to look for possible incompatibilities.

File: runtime/autoload/phpcomplete.vim [hunk 1]
```diff
@@ -2082,26 +2077,17 @@ function! phpcomplete#GetClassContentsStructure(file_path, file_lines, class_nam
...
-
-	silent! below 1new
-	silent! 0put =cfile
-	call search('\c\(class\|interface\|trait\)\_s\+'.a:class_name.'\(\>\|$\)')
-	let cfline = line('.')
-	call search('{')
-	let endline = line('.')
-
-	let content = join(getline(cfline, endline), "\n")
+	let popup_id = popup_create(a:file_lines, {'hidden': v:true})
```
[the remark this answers is not in the document]
Comment (mhinz) [reply runtime/autoload/phpcomplete.vim:2083]: Probably not. `popup*` is N/A and `win_execute()` is also not merged yet.

That will happen more often in the future and I guess it's infeasable to port all of them to Nvim syntax. For the occasional issue where someone complains about differences between Nvim and Vim in such cases, we should just point to LSP instead, which is supposed to offer a more complete experience anyway.

File: runtime/autoload/zip.vim [hunk 2]
```diff
@@ -65,10 +65,10 @@ endif
 " zip#Browse: {{{2
 fun! zip#Browse(zipfile)
 "  call Dfunc("zip#Browse(zipfile<".a:zipfile.">)")
-  " sanity check: ensure that the zipfile has "PK" as its first two letters
+  " sanity check: insure that the zipfile has "PK" as its first two letters
   "               (zipped files have a leading PK as a "magic cookie")
   if !filereadable(a:zipfile) || readfile(a:zipfile, "", 1)[0] !~ '^PK'
-   exe "noautocmd e ".fnameescape(a:zipfile)
+   exe "noswapfile noautocmd noswapfile e ".fnameescape(a:zipfile)
```
[the remark this answers is not in the document]
Comment (janlazo) [reply runtime/autoload/zip.vim:71]: https://github.com/vim/vim/commit/2963456ff2b740244b3a064785fe681b1998d75e#diff-c3431e35dce9ef0098df28d227c4539fc744a55e32dca0291be3eeddab0fe245R71

File: runtime/autoload/phpcomplete.vim (hunk 1 above)
[the remark this answers is not in the document]
Comment (janlazo) [reply runtime/autoload/phpcomplete.vim:2083]: Unsetting php omnifunc should be enough to avoid this code for now.

**Comment by mhinz** [comment]
**`win_complete()`**

- `runtime/autoload/phpcomplete.vim`: It's used all over the place together with `popup*()`, so don't merge.
- `runtime/doc/usr_41.txt`: Probably fine if we intend to merge `win_complete()` anyway in the future.
- `src/nvim/testdir/test_conceal.vim`: Comment it out for now.

**Comment by mhinz** [comment]
General question: Will we ever change `v:version` again? Currently it's 800.

`runtime/autoload/netrw.vim` for example checks if it can use `popup_beval()`/`popup_close()` like this:

```
v:version > 802 || (v:version == 802 && has("patch486"))
```

So, unless we satisfy that condition, we should be good (for now).

**Comment by jamessan** [comment]
> General question: Will we ever change v:version again? Currently it's 800.

Yes, when we've finished porting the 8.0 patches.

File: runtime/autoload/phpcomplete.vim (hunk 1 above)
[the remark this answers is not in the document]
Comment (janlazo) [reply runtime/autoload/phpcomplete.vim]: Got no upvotes so I assume that's a no.

How about adding the vim7 branch of https://github.com/shawncplus/phpcomplete.vim in `pack/` and adding a plugin to load it by default?

[the remark this answers is not in the document]
Comment (janlazo) [reply runtime/autoload/phpcomplete.vim]: I guess not.

I reverted the patch so this should be resolved for now.

File: runtime/autoload/zip.vim (hunk 2 above)
[the remark this answers is not in the document]
Comment (janlazo) [reply runtime/autoload/zip.vim:71]: Yes.

**Comment by janlazo** [comment]
runtime/autoload/phpcomplete.vim is reverted.

win_execute PR https://github.com/neovim/neovim/pull/13664

src/nvim/testdir/test_conceal.vim passes because it doesn't run `win_execute`. No need to comment out dead code. Dead code is kept for merging.

**Comment by janlazo** [comment]
I'm merging this since the reviewer comments are handled here or in a separate PR (ie. win_execute). 
Blame me if anything breaks. :runner:

**Comment by janlazo** [comment]
As for the vi_diff.txt, I'll fix it up in a separate PR. There are unmerged vim patches that shift the Vi references to vi_diff.txt

**Pull request merged**